cache/lru_cache_test.cc

```diff
@@ -547,6 +566,195 @@ TEST_F(LRUSecondaryCacheTest, FullCapacityTest) {
   cache.reset();
   secondary_cache.reset();
 }
+
+TEST_F(DBSecondaryCacheTest, TestSecondaryCacheCorrectness) {
+  LRUCacheOptions opts(4 * 1024, 0, false, 0.5, nullptr,
+                       kDefaultToAdaptiveMutex, kDontChargeCacheMetadata);
+  std::shared_ptr<TestSecondaryCache> secondary_cache(
+      new TestSecondaryCache(2048 * 1024));
+  opts.secondary_cache = secondary_cache;
+  std::shared_ptr<Cache> cache = NewLRUCache(opts);
+  BlockBasedTableOptions table_options;
+  table_options.block_cache = cache;
+  table_options.block_size = 4 * 1024;
+  Options options = GetDefaultOptions();
+  options.create_if_missing = true;
+  options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  DestroyAndReopen(options);
+  Random rnd(301);
+  const int N = 6;
+  for (int i = 0; i < N; i++) {
+    std::string p_v = rnd.RandomString(1000);
+    ASSERT_OK(Put(Key(i), p_v));
+  }
+  ASSERT_OK(Flush());
+  Compact("a", "z");
+  // After compaction, a iterator is created to test if the file is correct, so
```

Comment:
Does compaction also create an iterator to test the file? I thought only flush did that.